behaviour/graql/language/define.feature

```diff
@@ -1018,92 +928,700 @@ Feature: Graql Define Query
-      | NAL | label | number-of-artificial-limbs |
+      |     | check | value             |
+      | JSE | label | john-smiths-email |
+      | RUL | label | rule              |
     Then uniquely identify answer concepts
       | x   |
-      | NOL |
-      | NAL |
+      | JSE |
+      | RUL |
 
 
-  Scenario: define a type as abstract errors if has non-abstract parent types (?)
+  Scenario: define a rule with no `when` clause throws
+    Then graql define throws
+      """
+      define
+      nickname sub name;
+      person has nickname;
+      has-nickname-bob sub rule,
+      then {
+        $p has nickname "Bob";
+      };
+      """
 
 
-  #########
-  # RULES #
-  #########
+  Scenario: define a rule with no `then` clause throws
+    Then graql define throws
+      """
+      define
+      nickname sub name;
+      person has nickname;
+      robert sub rule,
+      when {
+        $p has name "Robert";
+      };
+      """
 
-  Scenario: define a rule creates a rule
 
-  Scenario: define a rule with nested negation throws on commit
+  Scenario: define a rule with an empty `when` clause throws
+    Then graql define throws
+      """
+      define
+      nickname sub name;
+      person has nickname;
+      has-nickname-bob sub rule,
+      when {
+      }, then {
+        $p has nickname "Bob";
+      };
+      """
 
-  Scenario: define a rule with two conclusions throws on commit
 
-  Scenario: define a rule with disjunction throws on commit
+  Scenario: define a rule with an empty `then` clause throws
+    Then graql define throws
+      """
+      define
+      nickname sub name;
+      person has nickname;
+      robert sub rule,
+      when {
+        $p has name "Robert";
+      }, then {
+      };
+      """
 
-  Scenario: define rule with an unbound variable in the `then` throws on commit
 
-  Scenario: define a non-insertable `then` throws on commit (eg. missing specific roles, or attribute value)
+  Scenario: define a rule with negation creates a rule
+    Given graql define
+      """
+      define
+      only-child sub attribute, value boolean;
+      siblings sub relation, relates sibling;
+      person plays sibling, has only-child;
+      only-child-rule sub rule,
+      when {
+        $p isa person;
+        not {
+          (sibling: $p, sibling: $p2) isa siblings;
+        };
+      }, then {
+        $p has only-child true;
+      };
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x sub rule; get;
+      """
+    Then concept identifiers are
+      |     | check | value           |
+      | ONL | label | only-child-rule |
+      | RUL | label | rule            |
+    Then uniquely identify answer concepts
+      | x   |
+      | ONL |
+      | RUL |
+
+
+  # TODO: better scenario description? what exactly does 'unbound negated pattern variable' mean?
+  Scenario: define a rule with unbound negated pattern variables throws
```

Comment:
your TODO comment - all cleared up? "unbound negated pattern variable" means that the negation contains at least one variable that is bound outside of the negation sub-clause as well

Reply:
Hmm, I don't see any variable in this negation that is bound outside of the negation block - do you mean that the negation **must** contain at least one variable that was bound outside of the negation block?

Reply:
yes - MUST

Reply:
Renamed to `define a rule with a negation block whose pattern variables are all unbound outside the negation block throws`